setupTests-integration.js

```diff
@@ -41,7 +46,40 @@ jest.mock(
 beforeAll(async () => {
   // Initializing the Transaction Logger with the test name
   const testName = getTestNameFromGlobalJasmineInstance();
-  const testLogger = new TxLogger(testName);
-  testLogger.init();
+  const testLogger = new LoggerUtil(testName);
+  testLogger.init({ filePrettyPrint: true });
   loggers.test = testLogger;
+
+  // Initializing wallet benchmark logger
+  const walletBenchmarkLog = new LoggerUtil(
+    'wallet-benchmark',
+    { reusableFilename: true }
+  );
+  walletBenchmarkLog.init();
+  loggers.walletBenchmark = walletBenchmarkLog;
+
+  // Initializing transaction benchmark logger
+  const txBenchmarkLog = new LoggerUtil(
+    'tx-benchmark',
+    { reusableFilename: true }
+  );
+  txBenchmarkLog.init();
+  loggers.txBenchmark = txBenchmarkLog;
+
+  // Loading pre-calculated wallets
+  precalculationHelpers.test = new WalletPrecalculationHelper('./tmp/wallets.json');
+  await precalculationHelpers.test.initWithWalletsFile();
+});
+
+afterAll(async () => {
+  // Calculating wallets benchmark summary
+  await WalletBenchmarkUtil.logResults();
+
+  // Calculating transactions benchmark summary
+  const txSummary = TxBenchmarkUtil.calculateSummary();
+  loggers.test.insertLineToLog('Transaction summary', { txSummary });
+  await TxBenchmarkUtil.logResults();
+
+  // Storing data about used precalculated wallets
+  await precalculationHelpers.test.storeDbIntoWalletsFile();
```

Comment:
Each test suite is completely separated from the others, there is no way to share objects or memory between them.

The easiest way I found of keeping track of the pre-calculated wallet seeds already used was via the filesystem. In the same test run, each suite fetches data from the filesystem, changes it and writes back the updates for the next suites to use.
At the end of the test run, the file is uploaded as an artifact to GitHub for eventual debugging.

I updated line line 83's comment to clarify this a bit on [ca7be6f](https://github.com/HathorNetwork/hathor-wallet-headless/pull/189/commits/ca7be6f3f23be1ac25a091d2d1a37798dc98e621).

Reply:
> Each test suite is completely separated from the others, there is no way to share objects or memory between them.

Why do you want to share objects or memory among them?

All tests are supposed to be independent.

Reply:
Every call to `WalletHelper.getPrecalculatedWallet(walletId)` expects a pristine wallet without transactions on any of its addresses. Before this PR this was achieved by generating random 24-word seeds for each wallet, but now we only have a limited pool of pre-calculated wallets to pick from.

So, each time we pick one of those wallets and start it, we mark it as used, so that the next test suites do not try to initialize it again and find it with a transaction history on the privatenet.

Even though the tests are the least coupled we could make them, they all still rely on the same blockchain instance via the instantiated fullnode and hence share the same resource that we need to keep track of.

Reply:
But if you run the tests twice, then the wallets file will be the same? And wallets used in the first run won't be used in the second? Or you re-create this wallets file every new run?

Reply:
The 100 pre-calculated wallets we currently have are on the [`__tests__/integration/configuration/precalculated-wallets.json`](https://github.com/HathorNetwork/hathor-wallet-headless/blob/perf/integration-tests/__tests__/integration/configuration/precalculated-wallets.json) file.

On the `package.json`'s `test_network_up` script we copy this file to `./tmp/wallets.json` and the jest run picks up from there.

When we are on the CI environment, the filesystem is empty just like the privatenet blockchain: we have a guarantee both are in sync.

When the tests are running on the developer's machine, it's up to the developer to make sure the blockchain and the `tmp/wallets.json` file are in sync. The easiest way to confirm this is to:
- Run `npm run test_network_down` ( because the fullnode on the container is currently using memory storage, we wipe the blockchain history )
- Clear the temporary files folder: `rm -rf ./tmp` ( this will also ensure the benchmarks will be correctly written )

If the developer wants to keep the current state of the blockchain and the used wallets, just use `npm run test_network_integration` and it will keep both the temporary file and the blockchain unchanged from the last tests.

Maybe a new `.md` file should be created to explain the current use of the integration tests?

Reply:
I feel we should add `rm ./tmp/wallets.json` to the `test_network_down`, so it ends everything started in the `test_network_up`.

I guess the `.md` would be valid and maybe we could have a better strategy in order to allow even the blockchain to be used from 0 in every test but we can discuss this later.

Reply:
Good point! Added this command to the `package.json` script on [c7411f7](https://github.com/HathorNetwork/hathor-wallet-headless/pull/189/commits/c7411f761e664ff194c67efbb3e6f1e75938cfc7).